[BUG] ros2_medkit_linux_introspection fails on ROS buildfarm: missing pkg-config buildtool_depend

# Bug report

### Steps to reproduce
1. Run ROS buildfarm job `Hdev__ros2_medkit__ubuntu_jammy_amd64` (Humble on Ubuntu 22.04)
2. `rosdep install --from-paths src --ignore-src -r -y` installs declared dependencies only
3. `colcon build` proceeds through packages until it reaches `ros2_medkit_linux_introspection`

### Expected behavior

`ros2_medkit_linux_introspection` configures and builds successfully.

### Actual behavior

CMake configure fails because `pkg-config` is not installed on the build agent:

```
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindPkgConfig.cmake:99 (find_package_handle_standard_args)
  CMakeLists.txt:46 (find_package)

Configuring incomplete, errors occurred!
```

Cascade: `ros2_medkit_linux_introspection` fails, which aborts `ros2_medkit_beacon_common` and `ros2_medkit_graph_provider` (running in parallel) and skips 3 additional packages.

### Root cause

`src/ros2_medkit_discovery_plugins/ros2_medkit_linux_introspection/CMakeLists.txt` uses `find_package(PkgConfig REQUIRED)` at line 46 to locate `libsystemd` via `pkg_check_modules`, but `package.xml` does not declare `pkg-config` as a `<buildtool_depend>`. `rosdep` only installs declared dependencies, so the `pkg-config` binary is missing on clean buildfarm agents. Local dev environments typically have `pkg-config` installed system-wide, which masks the issue.

### Environment
- ros2_medkit version: 0.4.0
- ROS 2 distro: Humble
- OS: Ubuntu 22.04 (Jammy)

### Fix

Add `<buildtool_depend>pkg-config</buildtool_depend>` to `src/ros2_medkit_discovery_plugins/ros2_medkit_linux_introspection/package.xml`.

--- a/src/ros2_medkit_discovery_plugins/ros2_medkit_linux_introspection/package.xml
+++ b/src/ros2_medkit_discovery_plugins/ros2_medkit_linux_introspection/package.xml
@@ -9,6 +9,7 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>ros2_medkit_cmake</buildtool_depend>
+  <buildtool_depend>pkg-config</buildtool_depend>
 
   <depend>ros2_medkit_gateway</depend>
   <depend>nlohmann-json-dev</depend>

